File: server/projects/profile/src/main/java/com/raxim/myscoutee/profile/repository/mongo/MessageRepository.java
package com.raxim.myscoutee.profile.repository.mongo;

import java.util.List;
import java.util.UUID;

import org.springframework.data.mongodb.repository.Aggregation;
import org.springframework.data.mongodb.repository.MongoRepository;
import org.springframework.data.repository.query.Param;

import com.raxim.myscoutee.profile.data.document.mongo.DBMessage;
import com.raxim.myscoutee.profile.data.dto.rest.MessageDTO;
import com.raxim.myscoutee.profile.data.dto.rest.PageParam;

public interface MessageRepository extends MongoRepository<DBMessage, UUID> {

    @Aggregation(pipeline = "findLastMessageByChannels")
    public List<MessageDTO> findLastMessageByChannels(@Param("param") PageParam param);

    public List<DBMessage> findByEventId(UUID eventId);

    @Aggregation(pipeline = "findMessagesByChannel")
    public List<MessageDTO> findMessagesByChannel(@Param("eventId") UUID eventId, @Param("param") PageParam param);
}

File: server/projects/profile/src/main/java/com/raxim/myscoutee/profile/repository/mongo/MessageRepository.kt
package com.raxim.myscoutee.profile.repository.mongo

import com.raxim.myscoutee.profile.data.document.mongo.Message
import org.springframework.data.mongodb.repository.MongoRepository
import org.springframework.data.mongodb.repository.Query
import org.springframework.stereotype.Repository
import java.util.*

@Repository
interface MessageRepository : MongoRepository<Message, UUID> {

    @Query("{lang: ?0}")
    fun findByLang(lang: String): Message?

}